### PR TITLE
My profile

### PR DESCRIPTION
**Pull request by franciscoghersi**

### CODE REVIEW

File: src/DonateTo.WebApplication/ClientApp/src/app/home/my-profile/my-profile/my-profile.component.css [hunk 1]
```diff
@@ -0,0 +1,17 @@
+.editable-cell {
+  position: relative;
+  padding: 5px 12px;
```
Comment (guillemurano) [inline src/DonateTo.WebApplication/ClientApp/src/app/home/my-profile/my-profile/my-profile.component.css:3]: Favor using ems

File: src/DonateTo.WebApplication/ClientApp/src/app/home/my-profile/my-profile/my-profile.component.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+<h3>{{ 'UserProfile.MyProfile' | translate }}</h3>
+<br />
+<nz-table #editRowTable nzBordered [nzData]="listOfRow" [nzFrontPagination]="false">
+  <thead>
+    <tr>
+      <th>{{ 'UserProfile.GeneralInformation' | translate }}</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr *ngFor="let data of editRowTable.data; let i = index" class="editable-row">
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/home/my-profile/my-profile/my-profile.component.html:11]: Don't use index. Replace the editable flag with a property flag of the object and change its value when it's clicked

`data.isEditable = false|true`

File: src/DonateTo.WebApplication/ClientApp/src/app/home/my-profile/my-profile/my-profile.component.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      <td>
+        <div class="editable-cell" [hidden]="editId === i" (click)="startEdit(i)">
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/home/my-profile/my-profile/my-profile.component.html:13]: Replace startEdit and stopEdit with a unique function called toggleEdit with the parameter "data". 
This function will change the current value of data.isEditable to the opposit value
`data.isEnabled = !data.isEnabled`

File: src/DonateTo.WebApplication/ClientApp/src/app/home/my-profile/my-profile/my-profile.component.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+export class MyProfileComponent implements OnInit {
+  listOfRow = [
+    { row: 'UserProfile.Name', value: 'Charly Brown' },
+    { row: 'UserProfile.Phone', value: '5472323456' },
+    { row: 'UserProfile.Country', value: 'United State' },
+    { row: 'UserProfile.State', value: 'Ohio' },
+    { row: 'UserProfile.Appartment', value: 'Columbus' },
+    { row: 'UserProfile.Street', value: 'National Rd SE 32' },
+    { row: 'UserProfile.PostalCode', value: '4212' },
+    { row: 'UserProfile.AdditionalInformation', value: 'N/A' },
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/home/my-profile/my-profile/my-profile.component.ts:17]: We should get the value from the backend

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/auth/auth.sandbox.ts [hunk 5]
```diff
@@ -45,6 +45,13 @@ export class AuthSandbox extends Sandbox {
     this.authService.logOut();
   }
 
+  /**
+   * Dispatches to the user profile
+   */
+  public myProfile(): void {
+    document.location.href = this.authService.redirectUri + '/my-profile';
```
[the remark this answers is not in the document]
Comment (Faculewis) [reply src/DonateTo.WebApplication/ClientApp/src/app/shared/auth/auth.sandbox.ts:52]: It should not be here

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/components/nav-menu/nav-menu.component.html [hunk 6]
```diff
@@ -15,6 +15,9 @@
     <li nz-menu-item *ngIf="!navMenuSandbox.isAuthenticated">
       <a (click)="login()"> {{ 'Auth.Login.Title' | translate }} </a>
     </li>
+    <li nz-menu-item *ngIf="navMenuSandbox.isAuthenticated">
+      <a (click)="myProfile()">{{ 'UserProfile.MyProfile' | translate }}</a>
```
[the remark this answers is not in the document]
Comment (Faculewis) [reply src/DonateTo.WebApplication/ClientApp/src/app/shared/components/nav-menu/nav-menu.component.html:19]: Yes, we have to use the routerlink directive:

`<a [routerLink]="['/my-profile']" >
  link to user component
</a>`

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/components/nav-menu/nav-menu.component.ts [hunk 7]
```diff
@@ -24,4 +24,8 @@ export class NavMenuComponent {
   logout() {
     this.navMenuSandbox.logout();
   }
+
+  myProfile() {
+    this.navMenuSandbox.myProfile();
+  }
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/components/nav-menu/nav-menu.component.ts:30]: Remove these lines

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/components/nav-menu/nav-menu.sandbox.ts [hunk 8]
```diff
@@ -40,6 +40,13 @@ export class NavMenuSandBox extends Sandbox {
     this.authSandbox.logout();
   }
 
+  /**
+   * Allows User go to their profile
+   */
+  public myProfile(): void {
+    this.authSandbox.myProfile();
+  }
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/components/nav-menu/nav-menu.sandbox.ts:48]: Remove these lines

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/containers/layout/donator-layout/donator-layout-routing.module.ts [hunk 9]
```diff
@@ -1,19 +1,23 @@
...
-        children: [
-            { path: '', component: HomeComponent}
-        ]
-    }
+  {
+    path: '',
+    component: DonatorLayoutComponent,
+    children: [
+      { path: '', component: HomeComponent },
+      { path: 'my-profile', component: MyProfileComponent },
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/containers/layout/donator-layout/donator-layout-routing.module.ts:14]: This should be in app-routing.module.ts and it should have the AuthGuard guard

File: src/DonateTo.WebApplication/ClientApp/i18n/en.json [hunk 10]
```diff
@@ -95,5 +95,21 @@
...
+    "Phone": "Phone",
+    "Country": "Country",
+    "State": "State",
+    "Apartment": "Apartment",
+    "Street": "Street",
+    "PostalCode": "Postal Code",
+    "AdditionalInformation": "Additional Information",
+    "OldPassword": "Old Password",
+    "NewPassword": "New Password",
+    "RepeatPassword": "Repeat New Password"
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/i18n/en.json:113]: Use the uppercase just for the first letter

File: src/DonateTo.WebApplication/ClientApp/i18n/es.json [hunk 11]
```diff
@@ -95,5 +95,21 @@
...
+    "Phone": "Teléfono",
+    "Country": "País",
+    "State": "Provincia",
+    "Apartment": "Localidad",
+    "Street": "Dirección",
+    "PostalCode": "Código Postal",
+    "AdditionalInformation": "Información Adicional",
+    "OldPassword": "Vieja Contraseña",
+    "NewPassword": "Nueva Contraseña",
+    "RepeatPassword": "Repetir Nueva Contraseña"
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/i18n/es.json:113]: Use the uppercase just for the first letter

File: src/DonateTo.WebApplication/ClientApp/src/app/app-routing.module.ts [hunk 12]
```diff
@@ -14,14 +14,19 @@ const appRoutes: Routes = [
...
+  },
   {
     path: '**',
     loadChildren: () => import('./basic-pages/basic-pages.module').then((m) => m.BasicPagesModule),
   },
 ];
 
 @NgModule({
-  imports: [RouterModule.forRoot(appRoutes)],
+  imports: [RouterModule.forRoot(appRoutes, { enableTracing: true })],
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/app-routing.module.ts:29]: Remove `{ enableTracing: true }`

File: src/DonateTo.WebApplication/ClientApp/src/app/home/my-profile/my-profile.component.html [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+          <th>{{ 'UserProfile.GeneralInformation' | translate }}</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr *ngFor="let data of editRowTable.data" class="editable-row">
+          <td>
+            <div class="editable-cell" [hidden]="isEnable == true" (click)="toggleEdit()">
+              {{ data.row | translate }}: {{ data.value }}
+            </div>
+            <input [hidden]="isEnable == false" type="text" nz-input [(ngModel)]="data.value" (blur)="toggleEdit()" />
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/home/my-profile/my-profile.component.html:19]: Change `isEnable == false` with `!isEnable`

File: src/DonateTo.WebApplication/ClientApp/src/app/home/my-profile/my-profile.component.html [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+    <nz-table #editRowTable nzBordered [nzData]="listOfRow" [nzFrontPagination]="false">
+      <thead>
+        <tr>
+          <th>{{ 'UserProfile.GeneralInformation' | translate }}</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr *ngFor="let data of editRowTable.data" class="editable-row">
+          <td>
+            <div class="editable-cell" [hidden]="isEnable == true" (click)="toggleEdit()">
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/home/my-profile/my-profile.component.html:16]: Change `isEnable == true` with `isEnable`

File: src/DonateTo.WebApplication/ClientApp/src/app/home/my-profile/my-profile.component.html [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+    <nz-table #newPassRowTable nzBordered [nzData]="listChangePassword" [nzFrontPagination]="false">
+      <thead>
+        <tr>
+          <th>{{ 'UserProfile.ChangePassword' | translate }}</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr *ngFor="let data of newPassRowTable.data" class="editable-row">
+          <td>
+            <div class="editable-cell" [hidden]="isEnable == true" (click)="toggleEdit()">
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/home/my-profile/my-profile.component.html:36]: Change `isEnable == true` with `isEnable`

File: src/DonateTo.WebApplication/ClientApp/src/app/home/my-profile/my-profile.component.html [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+              {{ data.row | translate }}: {{ data.value }}
+            </div>
+            <input [hidden]="isEnable == false" type="text" nz-input [(ngModel)]="data.value" (blur)="toggleEdit()" />
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/home/my-profile/my-profile.component.html:39]: Change `isEnable == false` with `!isEnable`

**Pull request merged**